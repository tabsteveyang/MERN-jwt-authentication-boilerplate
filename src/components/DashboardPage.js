import React from 'react';

const DashboardPage = () => (
  <div>
    Dashboard page content
  </div>
);

export default DashboardPage;
