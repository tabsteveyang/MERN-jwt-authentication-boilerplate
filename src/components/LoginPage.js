import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import AuthChecker from '../routers/AuthChecker';

export class LoginPage extends React.Component{
    constructor(props){
        super(props);
    }
    state = {
        email: '',
        password: ''
    };
    onEmailChange = (e) => {
        let email = e.target.value;
        this.setState({email});
    };
    onPasswordChange = (e) => {
        let password = e.target.value;
        this.setState({password});
    };
    onSubmit = (e) => {
        e.preventDefault();
        this.props.startLogin(this.state);
    };
    render(){
        return(
           <div className="box-layout">
               <AuthChecker history={this.props.history} isPublic={true} />
               <div className="box-layout__box">
                    <h1 className="box-layout__title">MERN App</h1>
                        <form>
                            <div className="form-group">
                        	<label htmlFor="inputEmail">Account</label>
                        	<input 
                        	    type="email" 
                        	    className="form-control form-control-lg" 
                        	    aria-describedby="emailHelp" 
                        	    placeholder="Email as account"
                        	    value={this.state.email}
                        	    onChange={this.onEmailChange}
                        	 ></input>
                            </div>
                            <div className="form-group">
                        	<label htmlFor="inputPassword">Password</label>
                        	<input 
                        	    type="password" 
                        	    className="form-control form-control-lg"
                        	    placeholder="Password"
                        	    value={this.state.password}
                        	    onChange={this.onPasswordChange}
                        	></input>
                            </div>
                            <hr/>
                            <button 
                                className="btn btn-primary" 
                                onClick={this.onSubmit}
                                disabled={(this.state.email.length === 0 || this.state.password.length === 0) ? true : false}
                            >Confirm</button>
                            { this.props.auth.status === 'error' ? (
                                <span className="err-hint" role="alert">
			            Invalid account or password.
			        </span>
                            ) : (
                                <span></span>
                            )}
                        </form>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    auth: state.auth
});
const mapDispatchToProps = (dispatch) => ({
  startLogin: (data) => dispatch(startLogin(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
