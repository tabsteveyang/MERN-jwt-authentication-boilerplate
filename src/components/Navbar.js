import React from 'react';
import { Link } from 'react-router-dom';
import LogoutBtn from './utils/LogoutBtn';

export const Navbar = () => (
    <nav className="navbar navbar-dark bg-dark">
	 <Link className="navbar-brand" to="/dashboard">
	     Boilerplate
	 </Link>
         <ul className="navbar-nav mr-auto">
               <li className="nav-item">
         	 <Link className="nav-link" to="/admin/createusr">
         	     CreateUser
         	 </Link>
               </li>
         </ul>
         <LogoutBtn />
    </nav>
);

export default Navbar;
