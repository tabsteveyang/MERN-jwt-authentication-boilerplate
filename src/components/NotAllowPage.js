import React from 'react';
import { Link } from 'react-router-dom';

const NotAllowPage = () => (
  <div className='container'>
    <h3>
        You are not allow to visit this page - <Link to="/dashboard">Go to Dashboard</Link>
    </h3>
  </div>
);

export default NotAllowPage;
