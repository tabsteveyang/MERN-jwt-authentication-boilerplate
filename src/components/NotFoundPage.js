import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div className='container'>
    <h3>
        404 - <Link to="/">Go home</Link>
    </h3>
  </div>
);

export default NotFoundPage;
