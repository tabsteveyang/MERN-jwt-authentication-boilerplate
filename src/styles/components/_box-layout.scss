.box-layout {
  align-items: center;
  //background: url('/images/bg.jpg');
  background: url('//c1.staticflickr.com/3/2937/14472303464_2e8c00f87d_h.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, .15);
  border-radius: 3px;
  padding: $l-size $m-size;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
  text-align: center;
}
