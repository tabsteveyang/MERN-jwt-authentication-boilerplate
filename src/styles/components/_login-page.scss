.err-hint{
    font-size: 1rem;
    padding-left: 1rem;
    color: #dc3545;
}
