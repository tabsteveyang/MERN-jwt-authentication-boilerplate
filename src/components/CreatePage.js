import React from 'react';
import { connect } from 'react-redux';
import validator from 'validator';
import { startCreateUser, changeStatus } from '../actions/forms';
import Navbar from './Navbar';
import AuthChecker from '../routers/AuthChecker';

export class CreatePage extends React.Component{
    constructor(props){
        super(props);
    }
    state = {
        email: '',
        password: '',
        name: '',
        settings: {
            language: 'EN'
        },
        privilige: {
            isActivate: false,
            setting: false,
            create: false,
            isAdmin: false
        },
        error: '',
    };
    dirtyForm = () => {
        //change the status of the form reducer.
        if(this.props.form.editStatus === 'clean')
            this.props.changeStatus('dirty');
    };
    onSubmit = (e) => {
        e.preventDefault();
        this.props.startCreateUser(this.state);
    }
    onEmailChange = (e) => {
        this.dirtyForm();
        let email = e.target.value;
        this.setState({email});
        setTimeout(() => {this.checkForm('email')}, 0);
    };
    onNameChange = (e) => {
        this.dirtyForm();
        let name = e.target.value;
        this.setState({name});
        setTimeout(() => {this.checkForm('name')}, 0);
    };
    onPasswordChange = (e) => {
        this.dirtyForm();
        let password = e.target.value;
        this.setState({password});
        setTimeout(() => {this.checkForm('password')}, 0);
    };
    checkForm = (field) => {
        if(field === 'email' || field === 'all'){
            if(!validator.isEmail(this.state.email)){
                this.setState({error: 'Invalid email format.'});
                return;
            }
        }
        if(field === 'name' || field === 'all'){
            if(validator.isEmpty(this.state.name)){
                this.setState({error: "Name can't be empty."});
                return;
            }
            if(this.state.name.length < 6){
                this.setState({error: 'Name has to be more then 6 characters.'});
                return;
            }
        }
        if(field === 'password' || field === 'all'){
            if(validator.isEmpty(this.state.password)){
                this.setState({error: "Password can't be empty."});
                return;
            }
            if(this.state.password.length < 6){
                this.setState({error: 'Password has to be more then 6 characters.'});
                return;
            }
        }

        this.setState({error: ''});
    };
    onInputCheck = (e) => {
        this.dirtyForm();
        let target = e.target.name;
        let value = e.target.value;
        let privilige = {
            ...this.state.privilige,
        };
        privilige[target] = e.target.checked;
        this.setState({privilige});
    };
    onSettingsChange = (e) => {
        this.dirtyForm();
        let target = e.target.name;
        let settings = {
            ...this.state.settings,
        };
        settings[target] = e.target.value;
        this.setState({settings});
    };
    componentDidUpdate(){
        if(this.props.form.editStatus === 'success'){
            alert('submit success!');
            setTimeout(() => {
                this.props.history.push('/'); //redirect to the main page after submit the form successfully.
            }, 1000);
        }else if(this.props.form.editStatus === 'error'){
            setTimeout(() => {this.checkForm('all')}, 0);
            this.props.changeStatus('dirty');
        }
    }
    render(){
        return(
        <div className='container-fluid'>
            <AuthChecker history={this.props.history} isPublic={false} accessRequire={'isAdmin'}/>
            <Navbar />
            <div className="jumbotron">
                <h1 className="display-4">Create User</h1>
                <form>
                    <div className="form-group">
                        <label htmlFor="inputEmail">Email address</label>
                        <input 
                            type="email" 
                            className="form-control form-control-lg" 
                            aria-describedby="emailHelp" 
                            placeholder="Enter email"
                            value={this.state.email}
                            onChange={this.onEmailChange}
                         ></input>
                    </div>
                    <div className="form-group">
                        <label htmlFor="inputName">Name</label>
                        <input 
                            type="text" 
	                    className="form-control form-control-lg" 
	            	    aria-describedby="nameHelp" 
	            	    placeholder="Enter name"
                            value={this.state.name}
                            onChange={this.onNameChange}
	                ></input>
                    </div>
                    <div className="form-group">
                        <label htmlFor="inputPassword">Password</label>
                        <input 
 	            	    type="password" 
	            	    className="form-control form-control-lg"
	            	    placeholder="Password"
                            value={this.state.password}
                            onChange={this.onPasswordChange}
	                ></input>
                    </div>
                    <div className="form-group">
                        <label htmlFor="privilige">Priviliges</label>
	            	<div className="form-check">
                               <input 
                                   className="form-check-input" 
                                   type="checkbox"
                                   checked={this.state.privilige.isActivate}
                                   onChange={this.onInputCheck}
                                   name="isActivate"
                               ></input>
                               <label className="form-check-label" htmlFor="priviligeCheck1">
                                   isActivate
                               </label>
                            </div>
	            	<div className="form-check">
                               <input 
                                   className="form-check-input" 
                                   type="checkbox"
                                   checked={this.state.privilige.setting}
                                   onChange={this.onInputCheck}
                                   name="setting"
                               ></input>
                               <label className="form-check-label" htmlFor="priviligeCheck2">
                                   setting
                               </label>
                            </div>
	            	<div className="form-check">
                               <input 
                                   className="form-check-input" 
                                   type="checkbox"
                                   checked={this.state.privilige.create}
                                   onChange={this.onInputCheck}
                                   name="create"
                               ></input>
                               <label className="form-check-label" htmlFor="priviligeCheck3">
                                   create
                               </label>
                            </div>
	            	<div className="form-check">
                               <input 
                                   className="form-check-input" 
                                   type="checkbox"
                                   checked={this.state.privilige.isAdmin}
                                   onChange={this.onInputCheck}
                                   name="isAdmin"
                               ></input>
                               <label className="form-check-label" htmlFor="priviligeCheck4">
                                   isAdmin
                               </label>
                            </div>
                    </div>
                    <div className="form-group">
                        <label htmlFor="form-control">Language</label>
                        <select 
                            className="form-control" 
                            name="language"
                            value={this.state.settings.language}
                            onChange={this.onSettingsChange}
                        >
			     <option>EN</option>
			</select>
                    </div>
                    <button 
                        className="btn btn-primary"
                        onClick={this.onSubmit}
                        disabled={this.state.error.length !== 0 || this.props.form.editStatus === 'clean'}
                    >Submit</button>
                        <div style={{height: '54px'}}>
                            <hr/>
                            {this.state.error !== '' ? 
                            <div className="alert alert-danger" role="alert">
                              {this.state.error}
                            </div>
                            : ''}
                        </div>
                </form>
            </div>
        </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    form: state.form
});
const mapDispatchToProps = (dispatch, props) => ({
    startCreateUser: (data) => dispatch(startCreateUser(data)),
    changeStatus: (status) => dispatch(changeStatus(status))
});

export default connect(mapStateToProps, mapDispatchToProps)(CreatePage);
