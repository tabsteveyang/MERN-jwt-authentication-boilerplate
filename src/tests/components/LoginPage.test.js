import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage auth={''}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const preventDefault = jest.fn();
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} auth={''} />);
  wrapper.find('button').simulate('click', {preventDefault});
  expect(startLogin).toHaveBeenCalled();
});
