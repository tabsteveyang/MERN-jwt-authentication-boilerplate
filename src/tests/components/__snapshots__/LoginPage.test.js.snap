// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <Connect(AuthChecker)
    isPublic={true}
  />
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      MERN App
    </h1>
    <form>
      <div
        className="form-group"
      >
        <label
          htmlFor="inputEmail"
        >
          Account
        </label>
        <input
          aria-describedby="emailHelp"
          className="form-control form-control-lg"
          onChange={[Function]}
          placeholder="Email as account"
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control form-control-lg"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <hr />
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
      >
        Confirm
      </button>
      <span />
    </form>
  </div>
</div>
`;
